### PR TITLE
Docs: Stack guide with GCP

### PR DESCRIPTION
**Pull request by suskin**
### Description of your changes

Related to #779 

As part of the 0.3 release, we want to document how to use stacks for the crossplane website, and for anyone looking at the docs in the repository. This changeset adds a first pass at an overall guide document, with a first pass at the GCP-specific provider section. We'll also need Azure-specific and AWS-specific instructions, but this should at least allow us to start testing the documentation.

I am thinking that we'll review and merge this documentation, and then polish it and add the other provider guides in future changesets.

### Testing done

I have clicked around on a bunch of the links, and have been viewing the documents locally as described in [this changeset](https://github.com/crossplaneio/crossplaneio.github.io/pull/25).

More testing is needed to make sure everything works smoothly!

### CODE REVIEW

File: cluster/examples/workloads/kubernetes/wordpress/gcp/provider.yaml [hunk 1]
```diff
@@ -4,49 +4,17 @@ data:
...
-  namespace: crossplane-system
+  namespace: gcp
   name: gcp-provider-creds
 type: Opaque
 ---
 apiVersion: gcp.crossplane.io/v1alpha2
 kind: Provider
 metadata:
-  namespace: crossplane-system
+  namespace: gcp
```
Comment (hasheddan) [inline cluster/examples/workloads/kubernetes/wordpress/gcp/provider.yaml]: I believe that we are wanting to move this to being `gcp-infra-dev`

File: docs/stacks-guide-gcp.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+Since this is an infrastructure stack, we need to specify that it's
+cluster-scoped via `--cluster` flag.
+```
+kubectl create namespace gcp
+kubectl crossplane stack generate-install --cluster 'crossplane/stack-gcp:master' stack-gcp | kubectl apply --namespace gcp -f -
+```
+
+The namespace that we install the stack to is also the one where our
+managed GCP resources will reside. When a developer requests a resource
+by creating a **resource claim** in a namespace `mynamespace`, the
```
Comment (hasheddan) [inline docs/stacks-guide-gcp.md]: I believe we will want `mynamespace` to be something like `app-project1-dev`

File: docs/stacks-guide-gcp.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+```
+
+Now we need to tell Crossplane which resource classes should be used to
+satisfy our claims in that app namespace. We will create portable
+classes that have have reference to non-portable ones that we created
+earlier. In our claims, we can refer to those portable classes directly
+or label one as the default portable class to be used in claims that do
+not have class reference.
+
+> Portable classes are a way of referring to non-portable resource classes in other namespaces.
```
Comment (hasheddan) [inline docs/stacks-guide-gcp.md]: May need to update if we use terminology other than "non-portable" in user-facing docs.

File: docs/stacks-guide-gcp.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+
+```
+kubectl -n gcp get connection.servicenetworking.gcp.crossplane.io/example-connection -o custom-columns='NAME:.metadata.name,FIRST_CONDITION:.status.conditions[0].status,SECOND_CONDITION:.status.conditions[1].status'
+```
+
+## Configure Provider Resources
+
+Once we have the network set up, we also need to tell Crossplane how to
+satisfy WordPress's claims for a database and a Kubernetes cluster.
+The resource classes serve as template for the new claimswe make.
```
Comment (hasheddan) [inline docs/stacks-guide-gcp.md]: "claimswe" -> "claims we"

File: docs/stacks-guide-gcp.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+```
+kubectl -n gcp get connection.servicenetworking.gcp.crossplane.io/example-connection -o custom-columns='NAME:.metadata.name,FIRST_CONDITION:.status.conditions[0].status,SECOND_CONDITION:.status.conditions[1].status'
+```
+
+## Configure Provider Resources
+
+Once we have the network set up, we also need to tell Crossplane how to
+satisfy WordPress's claims for a database and a Kubernetes cluster.
+The resource classes serve as template for the new claimswe make. 
+The following resource classes allow the GKECluster and CloudSQL claims
```
Comment (hasheddan) [inline docs/stacks-guide-gcp.md]: I think this should be "allow the KubernetesCluster and MySQLInstance claims"

File: docs/stacks-guide.md [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+Now that we've got Crossplane set up and configured to use a cloud
+provider, we're ready to add support for creating WordPresses! We'll do
+this using a Crossplane Stack. For more information about stacks, see
+the [full Stack documentation][stack-docs].
+
+We can use the [Crossplane CLI][crossplane-cli] to install our stack which adds support for
+Wordpress:
+
+```
+kubectl crossplane stack install 'crossplane/sample-stack-wordpress:latest' 'sample-stack-wordpress'
```
Comment (hasheddan) [inline docs/stacks-guide.md]: We will need to make sure that the wordpress stack is installed in the same `namespace` as the portable classes created in each individual stack guide. We could either standardize on the namespace or just mention that it needs to be the same here. I used this to install wordpress stack into the namespace of my portable classes:

```
kubectl crossplane stack generate-install  'crossplane/sample-stack-wordpress:latest' 'sample-stack-wordpress' | kubectl apply --namespace app-project1-dev -f -
```

File: docs/stacks-guide.md [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+Now that Crossplane supports Wordpress creation, we can ask Crossplane
+to spin up a Wordpress for us. We can do this by creating a Kubernetes
+resource that our Wordpress stack will recognize:
+
+```
+cat > my-wordpress.yaml <<EOF
+apiVersion: wordpress.samples.stacks.crossplane.io/v1alpha1
+kind: WordpressInstance
+metadata:
+  name: my-wordpressinstance
```
Comment (hasheddan) [inline docs/stacks-guide.md]: The `WordpressInstance` will also need to be in the correct namespace. I used this:

```
cat > my-wordpress.yaml <<EOF
apiVersion: wordpress.samples.stacks.crossplane.io/v1alpha1
kind: WordpressInstance
metadata:
  name: my-wordpressinstance
  namespace: app-project1-dev
EOF

kubectl apply -f my-wordpress.yaml
```

File: docs/stacks-guide.md [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+### Wait
+
+The Wordpress can take a while to spin up, because behind the scenes
+Crossplane is creating all of its dependendencies, which is a database
+and Kubernetes cluster. To check the status, we can look at the
+resources that Crossplane is creating for us:
+
+```
+# The claim for the database
+kubectl get mysqlinstance
```
Comment (hasheddan) [inline docs/stacks-guide.md]: Might be useful to include namespaces on these as well if we standardize.

Comment (muvaf) [reply docs/stacks-guide.md]: @suskin we need to change the namespace `mynamespace` to `app-project1-dev` in the gcp guide as well if we want to standardize on that.

File: docs/stacks-guide.md [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+### Use
+
+Once everything has been created, the ip address for the Wordpress
+instance will show up in the [Crossplane
+KubernetesApplicationResource][kubernetesapplicationresource-docs]
+which represents the workload's service. Here's a way to watch for the
+ip:
+
+```
+kubectl get kubernetesapplicationresource -n wordpresses -o custom-columns='NAME:.metadata.name,NAMESPACE:.spec.template.metadata.namespace,KIND:.spec.template.kind,SERVICE-EXTERNAL-IP:.status.remote.loadBalancer.ingress[0].ip' --watch
```
Comment (muvaf) [inline docs/stacks-guide.md]: `-n wordpresses` won't work

File: docs/stacks-guide.md (hunk 8 above)
Comment (muvaf) [reply docs/stacks-guide.md]: @hasheddan @suskin we can just omit the namespace for portable classes as well actually. Use whatever is set on kubectl for wordpress stack, wordpress instance and portable classes

Comment (hasheddan) [reply docs/stacks-guide.md]: @muvaf definitely true. However, I think we are trying to standardize on these namespace patterns, and it is more explicit IMO. I could be persuaded though... cc @prasek

File: docs/stacks-guide.md (hunk 9 above)
Comment (suskin) [reply docs/stacks-guide.md]: Should be fixed

File: docs/stacks-guide.md (hunk 8 above)
Comment (suskin) [reply docs/stacks-guide.md]: This should be fixed. I went with the explicit approach so readers can see an example of how to specify a namespace they want.

File: docs/stacks-guide.md (hunk 7 above)
Comment (suskin) [reply docs/stacks-guide.md]: Should be fixed; I ended up using a slightly inconsistent way of specifying it, but we can always fix that in a future changeset.

File: docs/stacks-guide.md (hunk 6 above)
Comment (suskin) [reply docs/stacks-guide.md]: This should be fixed now

File: docs/stacks-guide-gcp.md (hunk 5 above)
Comment (suskin) [reply docs/stacks-guide-gcp.md]: Should be fixed; I went even more general and didn't us the kind names

File: docs/stacks-guide-gcp.md (hunk 4 above)
Comment (suskin) [reply docs/stacks-guide-gcp.md]: Should be fixed

File: cluster/examples/workloads/kubernetes/wordpress/gcp/provider.yaml (hunk 1 above)
Comment (suskin) [reply cluster/examples/workloads/kubernetes/wordpress/gcp/provider.yaml]: Let's keep this note and roll it into the next changeset

File: docs/stacks-guide-gcp.md (hunk 2 above)
Comment (suskin) [reply docs/stacks-guide-gcp.md]: Should be fixed

File: docs/stacks-guide-gcp.md (hunk 3 above)
Comment (suskin) [reply docs/stacks-guide-gcp.md]: Good point. Let's keep track of this and consider it for a future changeset

**Pull request merged**